content/posts/2021-11-21-tăng-tốc.md

```diff
@@ -0,0 +1,65 @@
+---
+template: post
+title: How to choose a third party package
+slug: choose-third-party-package
+draft: false
+date: 2021-11-21T04:30:19.737Z
+description: TBU
+category: blog
+tags:
+  - npm
+  - third party package
+socialImage: "/media/npm-start.jpg"
+---
+![Reinvent the wheel](/media/wheel.jpg)
+
+## Introduction
+
+A wise man said "Don't reinvent the wheel". It's true and it applies to software development. When developing a feature, I bet sometimes you were in a situation "I need a third party package". So, how to choose a library or package correctly? In the past, whenever I thought I need an external library, I just google it (e.g: react draw chart), choose a random package, then try it. If it works, I use it. But it turned out that the strategy was not very good. In this article, I am going to share some personal tips on choosing a javascript package. But these tips applied to any other language's library, framework, or even picking a language for your team.
+
+### 1. Fit your need
+![npm start](/media/npm-start.jpg)
+
+It sounds obvious that you should choose a solution that solves your problem. A package usually tries to solve a set of problems in general. But sometimes, your use case is particular to your situation, and the package might not fit. Please make sure the third party code you are going to install handles sufficient, not too much, not too over engineer your problem. Many times, [DIY](#do-it-yourself) is a good option
+
+### 2. Easy to adopt
+![Easy](/media/easy.jpg)
+
+In industry, you don't work alone. So, pick a solution that your team members are able to adapt to quickly. You don't want to choose a solution that you are the only one who can implement and maintain.
+
+### 3. Popularity
+![Popularity](/media/popular.jpg)
+
+It's usually true that a good solution is a popular one. If a solution is good, why isn't it popular? I measure the popularity of a javascript package by the number of github stars and the number of weekly downloads on [https://npmjs.com](https://npmjs.com). A package with a big community proves that it solves the problem for many people. It comes with the advantages of having better support when we run into an issue. The popular package has more maintainers, contributors and users, so it's more mature than the less popular ones. It usually has good documentation and tutorials, which helps you to easily adopt and use it correctly and efficiently.
```

Comment:
- If a solution is good, why isn't it popular? => Câu này nên cho vào ngoặc đơn.
- A package with a big community proves that => A package backed by big community shows that
- The popular package has more maintainers => The **more** popular package has more maintainers
- so it's more mature => so it's more complete